src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -117,66 +113,67 @@
                            (rand-nth [1 2 3 4])))
   ([depth children]
-     (if (zero? depth)
-       {(random-string 10) (random-fact-value)}
-       {(random-string 10) (zipmap (repeatedly children #(random-string 10))
-                                   (repeatedly children #(random-structured-fact (rand-nth (range depth))
-                                                                                 (rand-nth (range children)))))})))
+   {(random-string 10)
+    (if (zero? depth)
+      (random-fact-value)
+      (zipmap (repeatedly children #(random-string 10))
+              (repeatedly children #(random-structured-fact (rand-nth (range depth))
+                                                            (rand-nth (range children))))))}))
 
 (defn maybe-tweak-catalog
   "Slightly tweak the given catalog, returning a new catalog, `rand-percentage`
   percent of the time."
-  [catalog rand-percentage]
+  [rand-percentage catalog]
   (if (< (rand 100) rand-percentage)
     (rand-catalog-mutation catalog)
     catalog))
 
-(defn update-catalog [catalog]
-  (assoc catalog "producer_timestamp" (time/now)))
+(defn update-producer-timestamp
+  "Updates the producer_timestamp to be current."
+  [entity]
+  (assoc entity "producer_timestamp" (time/now)))
 
-(defn update-report-run-fields
+(defn update-report
   "configuration_version, start_time and end_time should always change
    on subsequent report submittions, this changes those fields to avoid
    computing the same hash again (causing constraint errors in the DB)"
   [report]
-  (assoc report
-         "configuration_version" (ks/uuid)
-         "start_time" (time/now)
-         "end_time" (time/now)
-         "producer_timestamp" (time/now)))
+  (-> report
+      (assoc "configuration_version" (kitchensink/uuid)
+             "start_time" (time/now)
+             "end_time" (time/now))
+      update-producer-timestamp))
 
 (defn randomize-map-leaf
-  "Randomizes a fact leaf based on a percentage provided with `rp`."
-  [rp leaf]
-  (if (< (rand 100) rp)
-    (cond
-     (string? leaf) (random-string (inc (rand-int 100)))
-     (integer? leaf) (rand-int 100000)
-     (float? leaf) (* (rand) (rand-int 100000))
-     (ks/boolean? leaf) (random-bool))
-    leaf))
+  "Randomizes a fact leaf/"
+  [leaf]
+  (cond
+    (string? leaf) (random-string (inc (rand-int 100)))
+    (integer? leaf) (rand-int 100000)
+    (float? leaf) (* (rand) (rand-int 100000))
+    (kitchensink/boolean? leaf) (random-bool)))
 
 (defn randomize-map-leaves
-  "Runs through a map and randomizes leafs based on a percentage provided with
-   `rp`."
-  [rp value]
-  (cond
-   (map? value)
-   (ks/mapvals (partial randomize-map-leaves rp) value)
+  "Runs through a map and randomizes and random percentage of leaves."
+  [value & [rand-percentage]]
+  (let [rand-perc (or rand-percentage (rand-int 100))]
+    (cond
+      (map? value)
+      (kitchensink/mapvals (partial randomize-map-leaves rand-perc) value)
```

Comment:
So I actually did that intentionally, I wasn't really sure what we wanted. If the goal is to have `rp` of factsets altered, then this is what we want. If we pass `rp` through I think we end up in a situate where `rp` of `rp` fact leaves get updated, that's fine just curious what we want.